### ISSUE TITLE
Debug-formatting errors

### ISSUE THREAD

**Issue by BatmanAoD** [rust-lang/rust-clippy#14373]
### What it does

`unnecessary_debug_formatting` triggers on `Path` or `OsStr`, but there are other types that should almost never be (explicitly) debug-formatted. In particular, anything that implements `std::error::Error` is probably intended to be formatted with `fmt::Display` rather than `fmt::Debug`.

This could either be included in `unnecessary_debug_formatting` or added as a separate lint.

### Advantage

Ensures that errors are formatted the way the error-author intended.

### Drawbacks

* May be difficult to detect incorrect specifiers in `tracing` macros, since those use a slightly different syntax from `format!` (permitting key-value pairs before a final formatted string)
* May have false positives if someone wants to intentionally use the `Debug` format for an error

### Example

```rust
let mut f = match fs::create_new("my_file.txt") {
    Ok(f) => f,
    Err(e) => {
        eprintln!("Could not create file 'my_file.txt': {e:?}");
        return;
    }
};
```

Could be written as:

```rust
let mut f = match fs::create_new("my_file.txt") {
    Ok(f) => f,
    Err(e) => {
        eprintln!("Could not create file 'my_file.txt': {e}");
        return;
    }
};
```
